sklearn/decomposition/tests/test_pca.py

```diff
@@ -260,11 +261,12 @@ def test_singular_values():
               random_state=rng).fit(X)
     apca = PCA(n_components=2, svd_solver='arpack',
                random_state=rng).fit(X)
-    rpca = PCA(n_components=2, svd_solver='randomized',
+    # Increase the number of power iterations to get greater accuracy in tests
+    rpca = PCA(n_components=2, svd_solver='randomized', iterated_power=40,
                random_state=rng).fit(X)
-    assert_array_almost_equal(pca.singular_values_, apca.singular_values_, 12)
-    assert_array_almost_equal(pca.singular_values_, rpca.singular_values_, 1)
-    assert_array_almost_equal(apca.singular_values_, rpca.singular_values_, 1)
+    assert_allclose(pca.singular_values_, apca.singular_values_, 12)
```

Comment:
`assert_allclose` and `assert_array_almost_equal` have different signatures, your are setting `rtol=12` here, not `decimal=12`

I know it was in the original code already, but using named arguments helps avoiding this.

Reply:
Good catch!